### DISCUSSION TITLE
Constructing DAGs with ASTs

### DISCUSSION THREAD

**Discussion by santoshkumarradha** [AgnostiqHQ/covalent#708] [Ideas]
For a while I am really curious to explore this idea. Can we construct our lattice by looking at lattice's AST ? What we are doing inside lattice is almost like a very basic compiler with certain instructions being allowed. Can we make it a bit more general and just look at the AST to simplify the creation of DAG and even support more features off it ?

**Comment by santoshkumarradha** [comment]
Maybe one could easily accommodate conditional flows as well as indefinite dynamic flows this way ?  🤯

**Comment by cjao** [comment]
A related goal, which may be important for the on-prem use case, is that `Lattice.build_graph` should be made to work with serialized inputs. Since sublattice graphs are currently built by the Covalent server, the server environment must satisfy a sublattice workflow's package dependencies. The server should really never unpickle anything and work entirely with encoded functions, inputs, and their metadata.
